### ISSUE TITLE
Problems with datastore

### ISSUE THREAD

**Issue by Gerardo11011** [aws-amplify/amplify-android#295]
I have been trying to initialize app sync according to the Amazon guide and I get the message that the connection could be made using the following code
```
AWSMobileClient.getInstance()
            .initialize(applicationContext, object : Callback<UserStateDetails> {
                override fun onResult(userStateDetails: UserStateDetails) {
                    try {

                        Amplify.addPlugin(AWSApiPlugin()) // If using remote model synchronization
                        Amplify.configure(applicationContext)
                        println("ApiQuickStart " + "All set and ready to go!")
                    } catch (e: Exception) {
                        println("ApiQuickStart ERROR " + e.message)
                        println("ERRROR ApiQuickStart")
                    }

                }

                override fun onError(e: Exception) {
                    println("ApiQuickStart Initialization error. " + e)
                }
            })

```

> ApiQuickStart All set and ready to go!

But when you try to initialize the service with datastore I get the following error message

**Code**

```
AWSMobileClient.getInstance()
            .initialize(applicationContext, object : Callback<UserStateDetails> {
                override fun onResult(userStateDetails: UserStateDetails) {
                    try {
                        val dataStorePlugin = AWSDataStorePlugin.forModels(AmplifyModelProvider.getInstance())
                        Amplify.addPlugin(dataStorePlugin)
                        Amplify.addPlugin(AWSApiPlugin()) // If using remote model synchronization
                        Amplify.configure(applicationContext)
                        println("ApiQuickStart " + "All set and ready to go!")
                    } catch (e: Exception) {
                        println("ApiQuickStart ERROR " + e.message)
                        println("ERRROR ApiQuickStart")
                    }

                }

                override fun onError(e: Exception) {
                    println("ApiQuickStart Initialization error. " + e)
                }
            })

```

**Error**

> ApiQuickStart ERROR com.amplifyframework.datastore.DataStoreException: Error in initializing the SQLiteStorageAdapter

**My gradles**

```
apply plugin: 'com.android.application'

apply plugin: 'kotlin-android'

apply plugin: 'kotlin-android-extensions'

android {
    compileSdkVersion 29
    buildToolsVersion "29.0.3"
    defaultConfig {
        applicationId "com.android.example"
        minSdkVersion 15
        targetSdkVersion 29
        vectorDrawables.useSupportLibrary = true
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        multiDexEnabled = true
        android.defaultConfig.vectorDrawables.useSupportLibrary = true
    }
    dataBinding {
        enabled = true
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }
    compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }
    kotlinOptions {
        jvmTarget = "1.8"
    }
}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation 'androidx.appcompat:appcompat:1.1.0'
    implementation 'androidx.core:core-ktx:1.2.0'
    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
    implementation 'com.google.android.material:material:1.1.0'
    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
    implementation 'com.amplifyframework:core:0.9.1'
    implementation 'com.amplifyframework:aws-datastore:0.9.1'
    implementation 'com.amplifyframework:aws-api:0.9.1'
    implementation 'androidx.multidex:multidex:2.0.1'
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'

    implementation "android.arch.navigation:navigation-fragment-ktx:$version_navigation"
    implementation "android.arch.navigation:navigation-ui-ktx:$version_navigation"
    implementation 'androidx.gridlayout:gridlayout:1.0.0'
    implementation 'androidx.lifecycle:lifecycle-extensions:2.2.0'

}
```

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {

    ext{
        ext.kotlin_version = '1.3.70'
        version_navigation = '1.0.0'
    }
    repositories {
        google()
        jcenter()
        mavenCentral()

    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.6.1'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        classpath 'com.android.tools.build:gradle:3.6.1'
        //classpath 'com.amplifyframework:amplify-tools-gradle-plugin-beta:0.1.0'
        classpath 'com.amplifyframework:amplify-tools-gradle-plugin:0.2.1'
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

//apply plugin: 'com.amplifyframework.amplifytools'

allprojects {
    repositories {
        google()
        jcenter()
        
    }
}

task clean(type: Delete) {
    delete rootProject.buildDir
}

```

**Comment by CompileConnected** [comment]
I'm having the same issue here.
And my app behave strangely that I never saw before.
Just white screen and there's no ANR even though it should be.

tested on my xiaomi redmi note 4 mtk, and xiaomi redmi 4a
current ide: android studio 3.6.1

**Comment by Computr0n** [comment]
I am also running into this issue and its blocking me from evaluating amplify for our companies apps. 

Any workarounds for this blocker at the time being?

**Comment by jamesonwilliams** [comment]
Hi @Gerardo11011 , @CompileConnected thanks for the report. Do you have the full stack trace for this?
```
com.amplifyframework.datastore.DataStoreException: Error in initializing the SQLiteStorageAdapter
```

**Comment by Computr0n** [comment]
I have the following stack trace:

```java
E/SQLiteLog: (1) near "index": syntax error in "CREATE TABLE IF NOT EXISTS Memory (id TEXT PRIMARY KEY NOT NULL, hearingAidMemoriesId TEXT, index TEXT NOT NULL, name TEXT NOT NULL, memoryLocationId TEXT, FOREIGN KEY (memoryLoc
E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: com.starkey.starkit, PID: 21509
    java.lang.RuntimeException: com.amplifyframework.datastore.DataStoreException: Error in initializing the SQLiteStorageAdapter
        at io.reactivex.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
        at io.reactivex.internal.observers.BlockingMultiObserver.blockingGet(BlockingMultiObserver.java:93)
        at io.reactivex.Single.blockingGet(Single.java:2835)
        at com.amplifyframework.datastore.AWSDataStorePlugin.configure(AWSDataStorePlugin.java:123)
        at com.amplifyframework.core.category.Category.configure(Category.java:75)
        at com.amplifyframework.core.Amplify.configure(Amplify.java:134)
        at com.amplifyframework.core.Amplify.configure(Amplify.java:109)
        at com.starkey.starkit.StarKitApplication$onCreate$1.invokeSuspend(StarKitApplication.kt:30)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
     Caused by: com.amplifyframework.datastore.DataStoreException: Error in initializing the SQLiteStorageAdapter
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$initialize$2$SQLiteStorageAdapter(SQLiteStorageAdapter.java:233)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$WqAYkvd1XmGhKBCbpGbU-kI_MdA.run(Unknown Source:6)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: android.database.sqlite.SQLiteException: near "index": syntax error (code 1 SQLITE_ERROR): , while compiling: CREATE TABLE IF NOT EXISTS Memory (id TEXT PRIMARY KEY NOT NULL, hearingAidMemoriesId TEXT, index TEXT NOT NULL, name TEXT NOT NULL, memoryLocationId TEXT, FOREIGN KEY (memoryLocationId) REFERENCES Location(id) ON DELETE CASCADE);
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:990)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:597)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
        at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:33)
        at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1869)
        at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1797)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageHelper.createTablesAndIndexes(SQLiteStorageHelper.java:184)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageHelper.onCreate(SQLiteStorageHelper.java:97)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:411)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:316)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$initialize$2$SQLiteStorageAdapter(SQLiteStorageAdapter.java:206)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$WqAYkvd1XmGhKBCbpGbU-kI_MdA.run(Unknown Source:6) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:923) 
```

**Comment by CompileConnected** [comment]
@jamesonwilliams first of all i'm really sorry
because i thought my problem same as @Gerardo11011 
Turn out it's the sync problem when initializing the amplify with adding datastore plugin.
This behavour doesn't happen when use only AwsApiPlugin.
So my solution was to run initialization amplify in IO thread, i'm using coroutine and kotlin

```
 CoroutineScope(Dispatchers.IO).launch {
            try {
                val dataStore = AWSDataStorePlugin.forModels(AmplifyModelProvider.getInstance())
                Amplify.addPlugin(dataStore)
                Amplify.addPlugin(AWSApiPlugin())
                Amplify.configure(applicationContext)

                RxBus.publish(RxBusEvent.AmplifyInitialization(true))
            } catch (e: Exception) {
                RxBus.publish(RxBusEvent.AmplifyInitialization(false, e.message))
            }
        }
```

I'm really hoping you guys update the documentation about the amplify initialization is async

**Comment by jamesonwilliams** [comment]
@Gerardo11011 @Computr0n @CompileConnected I believe that has been fixed in the 0.10.0 release. Please re-open if that's not the case.
